### PR TITLE
Creating Feedback Provider Docs

### PR DESCRIPTION
**Pull request by StevenBucher98**
# PR Summary

This is first pass at creating a feedback provider doc. This doc outlines how to create a feedback provider that triggers on success and errors in the commandline. The success case only triggers if the user is using an aliased command and will provide feedback that it is in fact aliased and include the full name of the command. The error case will trigger when there is an error in the cmd and suggest the user try using `Get-Help` or check out online docs.

## PR Checklist

- [x] **Descriptive Title:** This PR's title is a synopsis of the changes it proposes.
- [x] **Summary:** This PR's summary describes the scope and intent of the change.
- [x] **Contributor's Guide:** I have read the [contributors guide][contrib].
- [x] **Style:** This PR adheres to the [style guide][style].

[contrib]: https://learn.microsoft.com/powershell/scripting/community/contributing/overview
[style]: https://learn.microsoft.com/powershell/scripting/community/contributing/powershell-style-guide

### CODE REVIEW

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (StevenBucher98) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:37]: @sdwheeler I am not sure how to do an "alert" of some kind to also highlight that the user needs to have the `PSFeedbackProvider` experimental feature enabled for this to work.

Comment (sdwheeler) [reply reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:33]: Added

**Review by sdwheeler — CHANGES_REQUESTED** [review]
@StevenBucher98  I did a full editorial pass with numerous changes. There is one TODO item left for you. Add instructions on how to build, package, and import the module.

**Comment by sdwheeler** [comment]
@StevenBucher98 just waiting on sign-off from @daxian-dbw.

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:306]: You have `var commandLine = context.CommandLine;` above, which is the command line that caused the error.

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:309]: Better use [string interpolation](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated) instead.

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:310]: Same here.

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:312]: and here. Maybe other places as well, where you use `+` to concatenate strings.

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:304]:
```suggestion
    if (erroredCommand is null)
    {
```

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:301]:
```suggestion
if (target == FeedbackTrigger.Error)
{
```

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:326]:
```suggestion
The following code implements the methods necessary from the **ICommandPredictor** interface to add predictor behavior to your feedback
```

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:330]:
```suggestion
- `CanAcceptFeedback()` - This method returns a Boolean value that indicates whether the predictor accepts a specific type of feedback.
```

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:331]:
```suggestion
- `GetSuggestion()` - This method
```

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:334]:
```suggestion
- `OnCommandLineAccepted()` - This
```

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:342]: Use xml doc comments please

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:354]: same here.

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:357]:
```suggestion
public SuggestionPackage GetSuggestion(
    PredictionClient client,
    PredictionContext context,
    CancellationToken cancellationToken)
```

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:383]: same here

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:394]: I believe you don't need to implement them now. The interface has default implementation for them.

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:396]: Remove

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:425]: This requires the `PSSubsystemPluginModel` experimental feature to be enabled. But we should be good as the NOTE below explains it further.

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:446]: Please update the code accordingly based on the feedback above.

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:1]: The resolution of `-arch.png` and `-output.png` seems to be not good. Any chance to make them clearer?

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:444]: Would it be possible to show the prediction in a GIF?

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:6]: I think this article should also point to the https://github.com/PowerShell/command-not-found repo as a sample for developers. Also point to the module in gallery: https://www.powershellgallery.com/packages/command-not-found/0.1.0.

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (StevenBucher98) [reply reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:308]: Not sure what you mean?

Comment (StevenBucher98) [reply reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:308]: I want to show off the last error property of `context`, I figured this would be easier than trying to grab the cmd in the string of the commandline

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [reply reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md]: This one is not resolved. Please consider moving the link to .NET 8.0 to the `Prerequisites` section.

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [reply reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:420]: This was left un-resolved.

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:39]: This paragraph seems redundant with the `Prerequisites` section. Maybe just add the pointer to "Download .NET 8.0" to the `- Install .NET 8 SDK - 8.0-preview.3 or higher` bullet in that section.

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:187]: This comment can be removed.

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:244]:
```suggestion
if (target == FeedbackTrigger.Success)
{
```

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:286]:
```suggestion
returned by `GetCommand` contains full name of the aliased command.
```
The last sentence can be removed given that the xml doc comment of the method gives the same information.

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:258]:
```suggestion
            actions.Add($"{aliasedCmd} --> {commandString}");
```

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:333]:
```suggestion
    actions.Add($"Get-Help {erroredCommand}");
```

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:607]:
```suggestion
                    actions.Add($"{aliasedCmd} --> {commandString}");
```

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:609]:
```suggestion
                }
```

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:260]:
```suggestion
        }
```

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:593]:
```suggestion
        if (target == FeedbackTrigger.Success)
        {
```

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [reply reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md]: Ah, right. We are good here.

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:356]: The `region` directive should be removed in the code walk-through doc. Let's only keep it in the full code snippet in the end.

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

File: reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md
Comment (daxian-dbw) [inline reference/docs-conceptual/dev-cross-plat/create-feedback-provider.md:257]:
```suggestion
        }
```

**Review by daxian-dbw — APPROVED** [review]

**Review by sdwheeler — APPROVED** [review]

**Pull request merged**
